In [1]:
print("Unang Aktibidad")

Unang Aktibidad


In [2]:
pip install --upgrade pip

Note: you may need to restart the kernel to use updated packages.


In [3]:
pip install pandas

Note: you may need to restart the kernel to use updated packages.


In [4]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [5]:
birds = pd.read_csv('datasets//penguins.csv')

In [6]:
birds

rowid    species     island  bill_length_mm  bill_depth_mm  \
0        1     Adelie  Torgersen            39.1           18.7   
1        2     Adelie  Torgersen            39.5           17.4   
2        3     Adelie  Torgersen            40.3           18.0   
3        4     Adelie  Torgersen             NaN            NaN   
4        5     Adelie  Torgersen            36.7           19.3   
..     ...        ...        ...             ...            ...   
339    340  Chinstrap      Dream            55.8           19.8   
340    341  Chinstrap      Dream            43.5           18.1   
341    342  Chinstrap      Dream            49.6           18.2   
342    343  Chinstrap      Dream            50.8           19.0   
343    344  Chinstrap      Dream            50.2           18.7   

     flipper_length_mm  body_mass_g     sex  year  
0                181.0       3750.0    male  2007  
1                186.0       3800.0  female  2007  
2                195.0       3250.0  female  2007  
3                  NaN          NaN     NaN  2007  
4                193.0       3450.0  female  2007  
..                 ...          ...     ...   ...  
339              207.0       4000.0    male  2009  
340              202.0       3400.0  female  2009  
341              193.0       3775.0    male  2009  
342              210.0       4100.0    male  2009  
343              198.0       3775.0  female  2009  

[344 rows x 9 columns]

In [7]:
birds.sex

0        male
1      female
2      female
3         NaN
4      female
        ...  
339      male
340    female
341      male
342      male
343    female
Name: sex, Length: 344, dtype: object

In [8]:
birds.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   rowid              344 non-null    int64  
 1   species            344 non-null    object 
 2   island             344 non-null    object 
 3   bill_length_mm     342 non-null    float64
 4   bill_depth_mm      342 non-null    float64
 5   flipper_length_mm  342 non-null    float64
 6   body_mass_g        342 non-null    float64
 7   sex                333 non-null    object 
 8   year               344 non-null    int64  
dtypes: float64(4), int64(2), object(3)
memory usage: 24.3+ KB


what is the average body mass of a penguin

In [9]:
birds.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [10]:
birds.describe() .T

count         mean         std     min       25%      50%  \
rowid              344.0   172.500000   99.448479     1.0    86.750   172.50   
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                       75%     max  
rowid               258.25   344.0  
bill_length_mm       48.50    59.6  
bill_depth_mm        18.70    21.5  
flipper_length_mm   213.00   231.0  
body_mass_g        4750.00  6300.0  
year               2009.00  2009.0

In [11]:
birds['weight_kilo'] = np.round(birds['body_mass_g'] / 1000,1)

In [12]:
birds.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
weight_kilo           2
dtype: int64

In [13]:
birds['bill_depth_mm'] = birds['bill_depth_mm'].fillna(birds['bill_depth_mm'].mean())

In [14]:
birds.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        2
bill_depth_mm         0
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
weight_kilo           2
dtype: int64

In [15]:
birds['flipper_length_mm'] = birds['flipper_length_mm'].fillna(birds['flipper_length_mm'].mean())

In [16]:
birds['bill_length_mm'] = birds['bill_length_mm'].fillna(birds['bill_length_mm'].mean())

In [17]:
birds['weight_kilo'] = birds['weight_kilo'].fillna(birds['weight_kilo'].mean())

In [18]:
birds['sex'] = birds['sex'].fillna(birds['sex'].mode()[0])

In [19]:
birds['body_mass_g'] = birds['body_mass_g'].fillna(birds['body_mass_g'].mean())

In [20]:
birds.isnull().sum()

rowid                0
species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
weight_kilo          0
dtype: int64

In [21]:
birds['year'] = birds['year'].astype('category')

In [22]:
birds['species'] = birds['species'].astype('category')

In [23]:
birds['rowid'] = birds['rowid'].astype('category')

In [24]:
birds['island'] = birds['island'].astype('category')

In [25]:
birds['sex'] = birds['sex'].astype('category')

In [26]:
birds.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   rowid              344 non-null    category
 1   species            344 non-null    category
 2   island             344 non-null    category
 3   bill_length_mm     344 non-null    float64 
 4   bill_depth_mm      344 non-null    float64 
 5   flipper_length_mm  344 non-null    float64 
 6   body_mass_g        344 non-null    float64 
 7   sex                344 non-null    category
 8   year               344 non-null    category
 9   weight_kilo        344 non-null    float64 
dtypes: category(5), float64(5)
memory usage: 26.9 KB


In [27]:
birds.head(10)

rowid species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0     1  Adelie  Torgersen        39.10000       18.70000         181.000000   
1     2  Adelie  Torgersen        39.50000       17.40000         186.000000   
2     3  Adelie  Torgersen        40.30000       18.00000         195.000000   
3     4  Adelie  Torgersen        43.92193       17.15117         200.915205   
4     5  Adelie  Torgersen        36.70000       19.30000         193.000000   
5     6  Adelie  Torgersen        39.30000       20.60000         190.000000   
6     7  Adelie  Torgersen        38.90000       17.80000         181.000000   
7     8  Adelie  Torgersen        39.20000       19.60000         195.000000   
8     9  Adelie  Torgersen        34.10000       18.10000         193.000000   
9    10  Adelie  Torgersen        42.00000       20.20000         190.000000   

   body_mass_g     sex  year  weight_kilo  
0  3750.000000    male  2007     3.800000  
1  3800.000000  female  2007     3.800000  
2  3250.000000  female  2007     3.200000  
3  4201.754386    male  2007     4.202047  
4  3450.000000  female  2007     3.400000  
5  3650.000000    male  2007     3.600000  
6  3625.000000  female  2007     3.600000  
7  4675.000000    male  2007     4.700000  
8  3475.000000    male  2007     3.500000  
9  4250.000000    male  2007     4.200000

In [28]:
birds['species'] == "Chinstrap"

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341     True
342     True
343     True
Name: species, Length: 344, dtype: bool

In [29]:
birds[birds['species'] == "Chinstrap"].count()

rowid                68
species              68
island               68
bill_length_mm       68
bill_depth_mm        68
flipper_length_mm    68
body_mass_g          68
sex                  68
year                 68
weight_kilo          68
dtype: int64

In [30]:
birds[birds['species'] == "Adelie"].count()

rowid                152
species              152
island               152
bill_length_mm       152
bill_depth_mm        152
flipper_length_mm    152
body_mass_g          152
sex                  152
year                 152
weight_kilo          152
dtype: int64

In [31]:
birds[birds['species'] == "Gentoo"].count()

rowid                124
species              124
island               124
bill_length_mm       124
bill_depth_mm        124
flipper_length_mm    124
body_mass_g          124
sex                  124
year                 124
weight_kilo          124
dtype: int64

<h1>
    HOW MANY GENTOO SPECIES HAVE WEIGHT MORE THAN 3kg?
</h1>

In [32]:
birds[(birds['weight_kilo'] < 3) & (birds['species'] == "Gentoo")].count()

rowid                0
species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
weight_kilo          0
dtype: int64

<h1>
    HOW MANY GENTOO SPECIES HAVE WEIGHT LESS THAN 3kg?
</h1>

In [33]:
birds[(birds['weight_kilo'] > 3) & (birds['species'] == "Gentoo")].count()

rowid                124
species              124
island               124
bill_length_mm       124
bill_depth_mm        124
flipper_length_mm    124
body_mass_g          124
sex                  124
year                 124
weight_kilo          124
dtype: int64

<H1>DOES GENTOO EXCLUSIVELY LIVE ON JUST ONE ISLAND?</H1>

In [34]:
birds['species'] == 'Gentoo'

0      False
1      False
2      False
3      False
4      False
       ...  
339    False
340    False
341    False
342    False
343    False
Name: species, Length: 344, dtype: bool

In [35]:
birds[birds['species'] == 'Gentoo']

rowid species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152   153  Gentoo  Biscoe        46.10000       13.20000         211.000000   
153   154  Gentoo  Biscoe        50.00000       16.30000         230.000000   
154   155  Gentoo  Biscoe        48.70000       14.10000         210.000000   
155   156  Gentoo  Biscoe        50.00000       15.20000         218.000000   
156   157  Gentoo  Biscoe        47.60000       14.50000         215.000000   
..    ...     ...     ...             ...            ...                ...   
271   272  Gentoo  Biscoe        43.92193       17.15117         200.915205   
272   273  Gentoo  Biscoe        46.80000       14.30000         215.000000   
273   274  Gentoo  Biscoe        50.40000       15.70000         222.000000   
274   275  Gentoo  Biscoe        45.20000       14.80000         212.000000   
275   276  Gentoo  Biscoe        49.90000       16.10000         213.000000   

     body_mass_g     sex  year  weight_kilo  
152  4500.000000  female  2007     4.500000  
153  5700.000000    male  2007     5.700000  
154  4450.000000  female  2007     4.400000  
155  5700.000000    male  2007     5.700000  
156  5400.000000    male  2007     5.400000  
..           ...     ...   ...          ...  
271  4201.754386    male  2009     4.202047  
272  4850.000000  female  2009     4.800000  
273  5750.000000    male  2009     5.800000  
274  5200.000000  female  2009     5.200000  
275  5400.000000    male  2009     5.400000  

[124 rows x 10 columns]

In [94]:
birds['island'].value_counts()

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

In [37]:
Gentoo_Biscoe = birds[(birds['species'] == 'Gentoo') & (birds['island'] == 'Biscoe')]

In [38]:
Gentoo_Biscoe

rowid species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152   153  Gentoo  Biscoe        46.10000       13.20000         211.000000   
153   154  Gentoo  Biscoe        50.00000       16.30000         230.000000   
154   155  Gentoo  Biscoe        48.70000       14.10000         210.000000   
155   156  Gentoo  Biscoe        50.00000       15.20000         218.000000   
156   157  Gentoo  Biscoe        47.60000       14.50000         215.000000   
..    ...     ...     ...             ...            ...                ...   
271   272  Gentoo  Biscoe        43.92193       17.15117         200.915205   
272   273  Gentoo  Biscoe        46.80000       14.30000         215.000000   
273   274  Gentoo  Biscoe        50.40000       15.70000         222.000000   
274   275  Gentoo  Biscoe        45.20000       14.80000         212.000000   
275   276  Gentoo  Biscoe        49.90000       16.10000         213.000000   

     body_mass_g     sex  year  weight_kilo  
152  4500.000000  female  2007     4.500000  
153  5700.000000    male  2007     5.700000  
154  4450.000000  female  2007     4.400000  
155  5700.000000    male  2007     5.700000  
156  5400.000000    male  2007     5.400000  
..           ...     ...   ...          ...  
271  4201.754386    male  2009     4.202047  
272  4850.000000  female  2009     4.800000  
273  5750.000000    male  2009     5.800000  
274  5200.000000  female  2009     5.200000  
275  5400.000000    male  2009     5.400000  

[124 rows x 10 columns]

In [39]:
Gentoo_Dream = birds[(birds['species'] == 'Dream') & (birds['island'] == 'Biscoe')]

In [40]:
Gentoo_Dream

Empty DataFrame
Columns: [rowid, species, island, bill_length_mm, bill_depth_mm, flipper_length_mm, body_mass_g, sex, year, weight_kilo]
Index: []

In [41]:
Gentoo_Torgersen = birds[(birds['species'] == 'Torgersen') & (birds['island'] == 'Biscoe')]

In [42]:
Gentoo_Torgersen

Empty DataFrame
Columns: [rowid, species, island, bill_length_mm, bill_depth_mm, flipper_length_mm, body_mass_g, sex, year, weight_kilo]
Index: []

<h1>INSIGHT N0.1
<h2>Gentoo lives exclusively on Biscoe</h2>

In [43]:
Adelie_Biscoe = birds[(birds['species'] == 'Adelie') & (birds['island'] == 'Biscoe')]

In [44]:
Adelie_Biscoe

rowid species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
20     21  Adelie  Biscoe            37.8           18.3              174.0   
21     22  Adelie  Biscoe            37.7           18.7              180.0   
22     23  Adelie  Biscoe            35.9           19.2              189.0   
23     24  Adelie  Biscoe            38.2           18.1              185.0   
24     25  Adelie  Biscoe            38.8           17.2              180.0   
25     26  Adelie  Biscoe            35.3           18.9              187.0   
26     27  Adelie  Biscoe            40.6           18.6              183.0   
27     28  Adelie  Biscoe            40.5           17.9              187.0   
28     29  Adelie  Biscoe            37.9           18.6              172.0   
29     30  Adelie  Biscoe            40.5           18.9              180.0   
50     51  Adelie  Biscoe            39.6           17.7              186.0   
51     52  Adelie  Biscoe            40.1           18.9              188.0   
52     53  Adelie  Biscoe            35.0           17.9              190.0   
53     54  Adelie  Biscoe            42.0           19.5              200.0   
54     55  Adelie  Biscoe            34.5           18.1              187.0   
55     56  Adelie  Biscoe            41.4           18.6              191.0   
56     57  Adelie  Biscoe            39.0           17.5              186.0   
57     58  Adelie  Biscoe            40.6           18.8              193.0   
58     59  Adelie  Biscoe            36.5           16.6              181.0   
59     60  Adelie  Biscoe            37.6           19.1              194.0   
60     61  Adelie  Biscoe            35.7           16.9              185.0   
61     62  Adelie  Biscoe            41.3           21.1              195.0   
62     63  Adelie  Biscoe            37.6           17.0              185.0   
63     64  Adelie  Biscoe            41.1           18.2              192.0   
64     65  Adelie  Biscoe            36.4           17.1              184.0   
65     66  Adelie  Biscoe            41.6           18.0              192.0   
66     67  Adelie  Biscoe            35.5           16.2              195.0   
67     68  Adelie  Biscoe            41.1           19.1              188.0   
100   101  Adelie  Biscoe            35.0           17.9              192.0   
101   102  Adelie  Biscoe            41.0           20.0              203.0   
102   103  Adelie  Biscoe            37.7           16.0              183.0   
103   104  Adelie  Biscoe            37.8           20.0              190.0   
104   105  Adelie  Biscoe            37.9           18.6              193.0   
105   106  Adelie  Biscoe            39.7           18.9              184.0   
106   107  Adelie  Biscoe            38.6           17.2              199.0   
107   108  Adelie  Biscoe            38.2           20.0              190.0   
108   109  Adelie  Biscoe            38.1           17.0              181.0   
109   110  Adelie  Biscoe            43.2           19.0              197.0   
110   111  Adelie  Biscoe            38.1           16.5              198.0   
111   112  Adelie  Biscoe            45.6           20.3              191.0   
112   113  Adelie  Biscoe            39.7           17.7              193.0   
113   114  Adelie  Biscoe            42.2           19.5              197.0   
114   115  Adelie  Biscoe            39.6           20.7              191.0   
115   116  Adelie  Biscoe            42.7           18.3              196.0   

     body_mass_g     sex  year  weight_kilo  
20        3400.0  female  2007          3.4  
21        3600.0    male  2007          3.6  
22        3800.0  female  2007          3.8  
23        3950.0    male  2007          4.0  
24        3800.0    male  2007          3.8  
25        3800.0  female  2007          3.8  
26        3550.0    male  2007          3.6  
27        3200.0  female  2007          3.2  
28        3150.0  female  2007    

In [45]:
Adelie_Dream = birds[(birds['species'] == 'Adelie') & (birds['island'] == 'Dream')]

In [46]:
Adelie_Dream

rowid species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
30     31  Adelie  Dream            39.5           16.7              178.0   
31     32  Adelie  Dream            37.2           18.1              178.0   
32     33  Adelie  Dream            39.5           17.8              188.0   
33     34  Adelie  Dream            40.9           18.9              184.0   
34     35  Adelie  Dream            36.4           17.0              195.0   
35     36  Adelie  Dream            39.2           21.1              196.0   
36     37  Adelie  Dream            38.8           20.0              190.0   
37     38  Adelie  Dream            42.2           18.5              180.0   
38     39  Adelie  Dream            37.6           19.3              181.0   
39     40  Adelie  Dream            39.8           19.1              184.0   
40     41  Adelie  Dream            36.5           18.0              182.0   
41     42  Adelie  Dream            40.8           18.4              195.0   
42     43  Adelie  Dream            36.0           18.5              186.0   
43     44  Adelie  Dream            44.1           19.7              196.0   
44     45  Adelie  Dream            37.0           16.9              185.0   
45     46  Adelie  Dream            39.6           18.8              190.0   
46     47  Adelie  Dream            41.1           19.0              182.0   
47     48  Adelie  Dream            37.5           18.9              179.0   
48     49  Adelie  Dream            36.0           17.9              190.0   
49     50  Adelie  Dream            42.3           21.2              191.0   
84     85  Adelie  Dream            37.3           17.8              191.0   
85     86  Adelie  Dream            41.3           20.3              194.0   
86     87  Adelie  Dream            36.3           19.5              190.0   
87     88  Adelie  Dream            36.9           18.6              189.0   
88     89  Adelie  Dream            38.3           19.2              189.0   
89     90  Adelie  Dream            38.9           18.8              190.0   
90     91  Adelie  Dream            35.7           18.0              202.0   
91     92  Adelie  Dream            41.1           18.1              205.0   
92     93  Adelie  Dream            34.0           17.1              185.0   
93     94  Adelie  Dream            39.6           18.1              186.0   
94     95  Adelie  Dream            36.2           17.3              187.0   
95     96  Adelie  Dream            40.8           18.9              208.0   
96     97  Adelie  Dream            38.1           18.6              190.0   
97     98  Adelie  Dream            40.3           18.5              196.0   
98     99  Adelie  Dream            33.1           16.1              178.0   
99    100  Adelie  Dream            43.2           18.5              192.0   
132   133  Adelie  Dream            36.8           18.5              193.0   
133   134  Adelie  Dream            37.5           18.5              199.0   
134   135  Adelie  Dream            38.1           17.6              187.0   
135   136  Adelie  Dream            41.1           17.5              190.0   
136   137  Adelie  Dream            35.6           17.5              191.0   
137   138  Adelie  Dream            40.2           20.1              200.0   
138   139  Adelie  Dream            37.0           16.5              185.0   
139   140  Adelie  Dream            39.7           17.9              193.0   
140   141  Adelie  Dream            40.2           17.1              193.0   
141   142  Adelie  Dream            40.6           17.2              187.0   
142   143  Adelie  Dream            32.1           15.5              188.0   
143   144  Adelie  Dream            40.7           17.0              190.0   
144   145  Adelie  Dream            37.3           16.8              192.0   
145   146  Adelie  Dream            39.0           18.7              185.0   
146   147  Adelie  Dream  

In [47]:
Adelie_Torgersen = birds[(birds['species'] == 'Adelie') & (birds['island'] == 'Torgersen')]

In [48]:
Adelie_Torgersen

rowid species     island  bill_length_mm  bill_depth_mm  \
0       1  Adelie  Torgersen        39.10000       18.70000   
1       2  Adelie  Torgersen        39.50000       17.40000   
2       3  Adelie  Torgersen        40.30000       18.00000   
3       4  Adelie  Torgersen        43.92193       17.15117   
4       5  Adelie  Torgersen        36.70000       19.30000   
5       6  Adelie  Torgersen        39.30000       20.60000   
6       7  Adelie  Torgersen        38.90000       17.80000   
7       8  Adelie  Torgersen        39.20000       19.60000   
8       9  Adelie  Torgersen        34.10000       18.10000   
9      10  Adelie  Torgersen        42.00000       20.20000   
10     11  Adelie  Torgersen        37.80000       17.10000   
11     12  Adelie  Torgersen        37.80000       17.30000   
12     13  Adelie  Torgersen        41.10000       17.60000   
13     14  Adelie  Torgersen        38.60000       21.20000   
14     15  Adelie  Torgersen        34.60000       21.10000   
15     16  Adelie  Torgersen        36.60000       17.80000   
16     17  Adelie  Torgersen        38.70000       19.00000   
17     18  Adelie  Torgersen        42.50000       20.70000   
18     19  Adelie  Torgersen        34.40000       18.40000   
19     20  Adelie  Torgersen        46.00000       21.50000   
68     69  Adelie  Torgersen        35.90000       16.60000   
69     70  Adelie  Torgersen        41.80000       19.40000   
70     71  Adelie  Torgersen        33.50000       19.00000   
71     72  Adelie  Torgersen        39.70000       18.40000   
72     73  Adelie  Torgersen        39.60000       17.20000   
73     74  Adelie  Torgersen        45.80000       18.90000   
74     75  Adelie  Torgersen        35.50000       17.50000   
75     76  Adelie  Torgersen        42.80000       18.50000   
76     77  Adelie  Torgersen        40.90000       16.80000   
77     78  Adelie  Torgersen        37.20000       19.40000   
78     79  Adelie  Torgersen        36.20000       16.10000   
79     80  Adelie  Torgersen        42.10000       19.10000   
80     81  Adelie  Torgersen        34.60000       17.20000   
81     82  Adelie  Torgersen        42.90000       17.60000   
82     83  Adelie  Torgersen        36.70000       18.80000   
83     84  Adelie  Torgersen        35.10000       19.40000   
116   117  Adelie  Torgersen        38.60000       17.00000   
117   118  Adelie  Torgersen        37.30000       20.50000   
118   119  Adelie  Torgersen        35.70000       17.00000   
119   120  Adelie  Torgersen        41.10000       18.60000   
120   121  Adelie  Torgersen        36.20000       17.20000   
121   122  Adelie  Torgersen        37.70000       19.80000   
122   123  Adelie  Torgersen        40.20000       17.00000   
123   124  Adelie  Torgersen        41.40000       18.50000   
124   125  Adelie  Torgersen        35.20000       15.90000   
125   126  Adelie  Torgersen        40.60000       19.00000   
126   127  Adelie  Torgersen        38.80000       17.60000   
127   128  Adelie  Torgersen        41.50000       18.30000   
128   129  Adelie  Torgersen        39.00000       17.10000   
129   130  Adelie  Torgersen        44.10000       18.00000   
130   131  Adelie  Torgersen        38.50000       17.90000   
131   132  Adelie  Torgersen        43.10000       19.20000   

     flipper_length_mm  body_mass_g     sex  year  weight_kilo  
0           181.000000  3750.000000    male  2007     3.800000  
1           186.000000  3800.000000  female  2007     3.800000  
2           195.000000  3250.000000  female  2007     3.200000  
3           200.915205  4201.754386    male  2007     4.202047  
4           193.000000  3450.000000  female  2007     3.400000  
5           190.000000  3650.000000    male  2007     3.600000  
6           181.000000  3625.000000  female  2007     3.600000  
7           195.000000  4675.000000    male  2007     4.700000  
8           193.000000  3475.000000    male  2007     3.500000  
9           19

<h1>INSIGHT N0.2
<h2>Adelie lives on all three islands</h2>

In [49]:
Chinstrap_Biscoe = birds[(birds['species'] == 'Chinstrap') & (birds['island'] == 'Biscoe')]

In [50]:
Chinstrap_Biscoe

Empty DataFrame
Columns: [rowid, species, island, bill_length_mm, bill_depth_mm, flipper_length_mm, body_mass_g, sex, year, weight_kilo]
Index: []

In [51]:
Chinstrap_Dream = birds[(birds['species'] == 'Chinstrap') & (birds['island'] == 'Dream')]

In [52]:
Chinstrap_Dream

rowid    species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276   277  Chinstrap  Dream            46.5           17.9              192.0   
277   278  Chinstrap  Dream            50.0           19.5              196.0   
278   279  Chinstrap  Dream            51.3           19.2              193.0   
279   280  Chinstrap  Dream            45.4           18.7              188.0   
280   281  Chinstrap  Dream            52.7           19.8              197.0   
..    ...        ...    ...             ...            ...                ...   
339   340  Chinstrap  Dream            55.8           19.8              207.0   
340   341  Chinstrap  Dream            43.5           18.1              202.0   
341   342  Chinstrap  Dream            49.6           18.2              193.0   
342   343  Chinstrap  Dream            50.8           19.0              210.0   
343   344  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  weight_kilo  
276       3500.0  female  2007          3.5  
277       3900.0    male  2007          3.9  
278       3650.0    male  2007          3.6  
279       3525.0  female  2007          3.5  
280       3725.0    male  2007          3.7  
..           ...     ...   ...          ...  
339       4000.0    male  2009          4.0  
340       3400.0  female  2009          3.4  
341       3775.0    male  2009          3.8  
342       4100.0    male  2009          4.1  
343       3775.0  female  2009          3.8  

[68 rows x 10 columns]

In [53]:
Chinstrap_Torgersen = birds[(birds['species'] == 'Chinstrap') & (birds['island'] == 'Torgersen')]

In [54]:
Chinstrap_Torgersen

Empty DataFrame
Columns: [rowid, species, island, bill_length_mm, bill_depth_mm, flipper_length_mm, body_mass_g, sex, year, weight_kilo]
Index: []

<h1>INSIGHT N0.3
<h2>Chinstrap lives exclusively on Dream</h2>

<h1>QUESTION NO.4
<h2>DO ADELIE'S SPECIE AFFECTS THE WEIGHT BASED ON THE ISLAND IT LIVES</h2>

In [55]:
Adelie_Biscoe_W = np.round(Adelie_Biscoe['weight_kilo'].mean(),2)

In [56]:
Adelie_Biscoe_W

np.float64(3.72)

In [57]:
Adelie_Dream_W = np.round(Adelie_Dream['weight_kilo'].mean(),2)

In [58]:
Adelie_Dream_W

np.float64(3.69)

In [59]:
Adelie_Torgersen_W = np.round(Adelie_Torgersen['weight_kilo'].mean(),2)

In [60]:
Adelie_Torgersen_W

np.float64(3.71)

<h1>INSIGHT N0.4
<h2>THE WEIGHT OF THE ADELIE SPECIE DOES NOT IMPACT WETHER IT LIVES ON ANY OF THE THREE ISLANDS</h2>

In [61]:
Gentoo_Biscoe_W = np.round(Gentoo_Biscoe['weight_kilo'].mean(),1)

In [62]:
Gentoo_Biscoe_W

np.float64(5.1)

<h1>INSIGHT N0.5
<h2>THE AVERAGE WEIGHT OF THE GENTOO SPECIE IS HEAVIER IN TERMS OF ITS SPECIE</h2>

In [63]:
Chinstrap_Dream_W = np.round(Chinstrap_Dream['weight_kilo'].mean(),1)

In [64]:
Chinstrap_Dream_W

np.float64(3.7)

<h1>INSIGHT N0.6
<h2>THE AVERAGE WEIGHT OF THE CHINSTRAP SPECIE IN DREAM IS THE SAME FOR ADELIE IN TERMS OF ITS AVERAGE WEIGHT</h2>

<h1>INSIGHT N0.7
<h2>THE GENTOO SPECIE IN BISCOE IS THE HEAVIEST AMONG THE THREE SPECIES</h2>

<h2>HOW TO USE GROUPBY()</h2>

<OL>
    <LI>NAME THE GROUPBY() REFERENCE DATA FRAME</LI>
    <LI>CALL THE DATAFRAME</LI>
    <LI>.GROUPBY FUNCTION</LI>
    <LI>COLUMNS TO GROUP</LI>
    <LI>GROUP BY FUNCTION</LI>
</OL>

In [65]:
birds_agg1 = birds.groupby(['species', 'island']).size()

C:\Users\LAB4-StudentPC03\AppData\Local\Temp\ipykernel_15032\1511608000.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  birds_agg1 = birds.groupby(['species', 'island']).size()


In [66]:
birds_agg1

species    island   
Adelie     Biscoe        44
           Dream         56
           Torgersen     52
Chinstrap  Biscoe         0
           Dream         68
           Torgersen      0
Gentoo     Biscoe       124
           Dream          0
           Torgersen      0
dtype: int64

<h1>
    INSIGHT NO.1-3
</h1>
<h1>Island residence per specie</h1>
<ol>
    <li>There are scattered Adelie specie on each island</li>
    <li>There are 68 Chinstrap penguins exclusively on Dream</li>
    <li>There are 124 Gentoo penguins exclusively on Biscoe</li>
</ol>

In [67]:
birds_agg2 = np.round(birds.groupby(['species', 'island'])['weight_kilo'].mean(),1)

C:\Users\LAB4-StudentPC03\AppData\Local\Temp\ipykernel_15032\3335494670.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  birds_agg2 = np.round(birds.groupby(['species', 'island'])['weight_kilo'].mean(),1)


In [68]:
birds_agg2

species    island   
Adelie     Biscoe       3.7
           Dream        3.7
           Torgersen    3.7
Chinstrap  Biscoe       NaN
           Dream        3.7
           Torgersen    NaN
Gentoo     Biscoe       5.1
           Dream        NaN
           Torgersen    NaN
Name: weight_kilo, dtype: float64

<h1>
    INSIGHT NO.4-6
</h1>
<h1>Average weight of each penguin specie in each islands</h1>
<ol>
    <li>The Average weight of Adelie specie on all islands is 3.7 Kilos</li>
    <li>The Average weight of Chinstrap specie on Dream is also 3.7 Kilos</li>
    <li>The Average weight of Gentoo specie on Biscoe is 5.1 Kilos</li>
</ol>

<h1>
    INSIGHT NO.7
</h1>
<h1>The heaviest specie is Gentoo that lives exclusively on Biscoe </h1>

In [69]:
birds_agg3 = birds.groupby(['species', 'island','year','sex']).size()

C:\Users\LAB4-StudentPC03\AppData\Local\Temp\ipykernel_15032\2639247255.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  birds_agg3 = birds.groupby(['species', 'island','year','sex']).size()


In [70]:
birds_agg3.unstack()

sex                       female  male
species   island    year              
Adelie    Biscoe    2007       5     5
                    2008       9     9
                    2009       8     8
          Dream     2007       9    11
                    2008       8     8
                    2009      10    10
          Torgersen 2007       8    12
                    2008       8     8
                    2009       8     8
Chinstrap Biscoe    2007       0     0
                    2008       0     0
                    2009       0     0
          Dream     2007      13    13
                    2008       9     9
                    2009      12    12
          Torgersen 2007       0     0
                    2008       0     0
                    2009       0     0
Gentoo    Biscoe    2007      16    18
                    2008      22    24
                    2009      20    24
          Dream     2007       0     0
                    2008       0     0
                    2009       0     0
          Torgersen 2007       0     0
                    2008       0     0
                    2009       0     0

<h1>
    INSIGHT NO.8
</h1>
<h1>The oldest penguin in adelie is a male in Torgersen in 2007 </h1>

In [71]:
birds_agg4 = np.round(birds.groupby(['species', 'island','sex'])['weight_kilo'].mean(),1)

C:\Users\LAB4-StudentPC03\AppData\Local\Temp\ipykernel_15032\3798003495.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  birds_agg4 = np.round(birds.groupby(['species', 'island','sex'])['weight_kilo'].mean(),1)


In [72]:
birds_agg4

species    island     sex   
Adelie     Biscoe     female    3.4
                      male      4.1
           Dream      female    3.3
                      male      4.0
           Torgersen  female    3.4
                      male      4.0
Chinstrap  Biscoe     female    NaN
                      male      NaN
           Dream      female    3.5
                      male      3.9
           Torgersen  female    NaN
                      male      NaN
Gentoo     Biscoe     female    4.7
                      male      5.4
           Dream      female    NaN
                      male      NaN
           Torgersen  female    NaN
                      male      NaN
Name: weight_kilo, dtype: float64

<h1>
    INSIGHT NO.9-11
</h1>
<h1>Average weight per sex of each penguin specie in each islands</h1>
<ol>
    <li>The Average weight of Adelie specie on all islands for female is 3.4 Kilos and male is 4.1 Kilos</li>
    <li>The Average weight of Chinstrap specie on Dream for female is 3.5 Kilos and male is 3.9 Kilos</li>
    <li>The Average weight of Gentoo specie on Biscoe for female is 4.7 Kilos and male is 5.4 Kilos</li>
</ol>

<h1>Does the bill length of a penguin affects the bill depth as it goes long or short?</h1>

In [73]:
b_length_depth = birds['bill_length_mm'].corr(birds['bill_depth_mm'])

In [74]:
b_length_depth

np.float64(-0.23505287035553282)

<h1>
    INSIGHT NO.12
</h1>

<p>The bill length does not affect the bill depth of a penguin as it goes long or short</p>

<h1>Does the weight of the penguin affects its flipper length?</h1>

In [75]:
b_flipper_mass = birds['flipper_length_mm'].corr(birds['weight_kilo'])

In [76]:
b_flipper_mass

np.float64(0.8706550442461823)

<h1>INSIGHT NO.13</h1>

<p>The weight does affect the flipper length of the penguin as it goes longer</p>

<h1>Does the bill length affect the flipper length?</h1>

In [77]:
b_flipper_mass = birds['flipper_length_mm'].corr(birds['weight_kilo'])

In [78]:
b_flipper_mass

np.float64(0.8706550442461823)

<h1>INSIGHT NO.14</h1>

<p>The bill length does affect the flipper length of the penguin is it both gets longer</p>

<h1>Does the bill depth affect the weight of the penguin?</h1>

In [79]:
b_depth_mass = birds['bill_depth_mm'].corr(birds['weight_kilo'])

In [80]:
b_depth_mass

np.float64(-0.4707176291681673)

<h1>INSIGHT NO.15</h1>

<p>The bill depth does affect the weight of the penguin</p>

In [81]:
birds.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   rowid              344 non-null    category
 1   species            344 non-null    category
 2   island             344 non-null    category
 3   bill_length_mm     344 non-null    float64 
 4   bill_depth_mm      344 non-null    float64 
 5   flipper_length_mm  344 non-null    float64 
 6   body_mass_g        344 non-null    float64 
 7   sex                344 non-null    category
 8   year               344 non-null    category
 9   weight_kilo        344 non-null    float64 
dtypes: category(5), float64(5)
memory usage: 26.9 KB


In [82]:
pip install scipy

Note: you may need to restart the kernel to use updated packages.


In [85]:
from scipy.stats import f_oneway

<h2>requirements for one way anova</h2>
<h3>Filter columns and  its data</h3>

In [92]:
birds[birds['sex'] == 'male']['weight_kilo']

0      3.800000
3      4.202047
5      3.600000
7      4.700000
8      3.500000
         ...   
334    3.800000
336    4.000000
339    4.000000
341    3.800000
342    4.100000
Name: weight_kilo, Length: 179, dtype: float64

In [93]:
birds[birds['sex'] == 'female']['weight_kilo']

1      3.8
2      3.2
4      3.4
6      3.6
12     3.2
      ... 
335    3.5
337    3.6
338    3.6
340    3.4
343    3.8
Name: weight_kilo, Length: 165, dtype: float64

In [91]:
f_stats, p_value = f_oneway(birds[birds['sex']=='male']['weight_kilo'],
                          birds[birds['sex']=='female']['weight_kilo'])

print(p_value)

2.5288724247849435e-15


<h1>INSIGHT NO.16</h1>

<h2>There is a correlation between sex and body mass</h2>